src/Schema/Events/ImpressionTracking.ts

```diff
@@ -104,6 +104,28 @@ export interface ProgressiveOnboardingTooltipViewed {
   type: string
 }
 
+/**
+ * A user viewed a toast. `subject` will describe what toast it is.
+ *
+ * This schema describes events sent to Segment from [[viewedToast]]
+ *
+ *  @example
+ *  ```
+ *  {
+ *    action: "viewedToast",
+ *    context_module: "artworkRecommendations",
+ *    context_screen_owner_type: "artworkGrid",
+ *    subject: "Set an artwork budget in your profile at any time."
+ *  }
+ * ```
+ */
+export interface ViewedToast {
+  action: ActionType.viewedToast
+  context_module: ContextModule
+  context_screen_owner_type?: OwnerType
```

Comment:
I can imagine there would always be a screen associated with this view event. Maybe we make this required and update the example to use the home view?

I don't know if we would often have an additional context module as these toasts usually float above the screen you're on. Is that attribute necessary in this case?

Reply:
for the context, the usage will look like this:
```
...
      context_screen_owner_type: OwnerType.artworkRecommendations,
      context_module: ContextModule.artworkGrid,
...
```

Reply:
Gotcha. I think the example values are swapped at the moment. I see `artworkGrid` used for `context_screen_owner_type.

And I suppose `artworkGrid` makes sense as that's where the trigger logic is found – we fire the toast in response to scroll depth on the grid.